test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatAuthenticationTests.cs

```diff
@@ -10,14 +11,31 @@ namespace NuGet.XPlat.FuncTest
 {
     public class XPlatAuthenticationTests
     {
-        [Fact(Skip = "There are flaky servers in the config file")]
-        public void Restore_WithAuthenticatedSource_Succeeds()
+        [Theory]
+        // Try with config file in the project directory
+        [InlineData(TestServers.Artifactory, false)]
+        [InlineData(TestServers.Klondike, false)]
+        [InlineData(TestServers.MyGet, false)]
+        [InlineData(TestServers.Nexus, false)]
+        [InlineData(TestServers.NuGetServer, false)]
+        [InlineData(TestServers.ProGet, false)]
+        // Try with config file in a different directory
+        [InlineData(TestServers.Artifactory, true)]
+        [InlineData(TestServers.Klondike, true)]
+        [InlineData(TestServers.MyGet, true)]
+        [InlineData(TestServers.Nexus, true)]
+        [InlineData(TestServers.NuGetServer, true)]
+        [InlineData(TestServers.ProGet, true)]
+        public void Restore_WithAuthenticatedSource_Succeeds(string sourceUri, bool configInDifferentDirectory)
```

Comment:
I can rename the file and function, it's useful to try to restore against all of these servers.
Would that work?

---
In reply to: [63444223](https://github.com/NuGet/NuGet.Client/pull/596#discussion_r63444223) [](ancestors = 63444223)